modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/plugin/ArtifactPluginDescriptorFactoryTestCase.java

```diff
@@ -199,12 +203,31 @@ public File build() throws IOException {
         throw new IllegalStateException(String.format("File '%s' already exists", pluginProperties.getAbsolutePath()));
       }
 
+      pluginProperties.createNewFile();
       addDescriptorProperty(pluginProperties, EXPORTED_CLASS_PACKAGES_PROPERTY, this.exportedClassPackages);
       addDescriptorProperty(pluginProperties, EXPORTED_RESOURCE_PROPERTY, this.exportedResources);
-
+      addJsonDescriptor();
       return pluginProperties;
     }
 
+    private void addJsonDescriptor() {
```

Comment:
This code looks like ArtifactPluginFileBuilder#createMulePluginJsonResource. Refactor to remove duplication